port/port_stdcxx.h

```diff
@@ -200,6 +203,113 @@ inline bool Zstd_Uncompress(const char* input, size_t length, char* output) {
 #endif  // HAVE_ZSTD
 }
 
+inline bool Zlib_Compress(int level, const char* input, size_t length,
+                          std::string* output, uint8_t raw = 0) {
+#if HAVE_ZLIB
+  const size_t BUFSIZE = 128 * 1024;
+  unsigned char temp_buffer[BUFSIZE];
+  // reserve enough memory to not reallocate on the fly
+  output->reserve(output->size() + compressBound(length));
```

Comment:
do we need to take in consideration the output's size here or compressBound would be enough? 
```suggestion
  output->reserve(compressBound(length));
```

Reply:
I removed it as you suggested. `output` is passed in with a size of 0, so not sure why this was added in the first place.